### PR TITLE
advanced require

### PR DESCRIPTION
**Pull request by hrgdavor**
update require function to read files from modules, not just bundled modules on jsdelivr.

### CODE REVIEW

**Comment by hrgdavor** [comment]
this works for sqed folder example, and has a lot of console.log ...

I had to chenge things around a bit to work with how jsdelivr handles files, and alos add support fore require of `.json` .. and this can be seen as a gateway to supporting require `.stl`

File: packages/require/src/require.js [hunk 1]
```diff
@@ -107,14 +141,28 @@ export function require(urlOrSource, transform, _readFile, _base, root, readModu
...
+
     cache = requireCache[isRelativeFile ? 'local':'module']
     exports = cache[cacheUrl] // get from cache
     if (!exports) {
       // not cached
       try {
         source = readFile(url, { base })
+        if(url.includes('jsdelivr.net')){
+          // jsdelivr will read package.json and tell us what the main file is
+          const srch = ' * Original file: '
```
Comment (platypii) [inline packages/require/src/require.js:157]: This was the part I didn't quite know how to handle. Nice work

File: packages/require/src/require.js [hunk 2]
```diff
@@ -131,15 +179,23 @@ export function require(urlOrSource, transform, _readFile, _base, root, readModu
...
+    }
   }
-
-  if (typeof exports !== 'object') {
+  
+  // allow aslo function workaround for weid case in @jscad/io-utils/Blob.js
+  if (typeof exports !== 'object' && typeof exports !== 'function') {
+    console.error('exports',typeof exports, exports)
     throw new Error(`module ${cacheUrl} did not export an object`)
   }
```
Comment (platypii) [inline packages/require/src/require.js:200]: Technically, you could also do things like:
```js
module.exports = 'export a string'
module.exports = 3.14
```

What's interesting though is that if you load say, an empty file, then exports will actually be empty object `{}`. So I don't think this entire check is doing... anything? We handle the missing main case in `worker.js:51`. I propose we just delete this whole if block?

```suggestion
```

File: apps/jscad-web/build.js [hunk 3]
```diff
@@ -37,7 +37,7 @@ if(!(dev & existsSync(outDir + "/docs"))){
 /**************************** BUILD JS that is static *************/
 await buildBundle(outDir + '/build', 'bundle.threejs.js', { globalName: 'THREE' })
 await buildBundle(outDir + '/build', 'bundle.jscad_modeling.js', { format: 'cjs' })
-await buildBundle(outDir + '/build', 'bundle.jscad.io.js', { globalName: 'jscad_io' })
+await buildBundle(outDir + '/build', 'bundle.jscad_io.js', { format:'cjs' /*,globalName: 'jscad_io'*/ })
```
Comment (platypii) [inline apps/jscad-web/build.js:40]: nit: remove the commented out bit?

**Review by platypii — APPROVED** [review]
Really nice work! This probably fixes a ton of npm packages (not just jscad/io). And it's nice how fast it is after you added the bundle. :rocket:

**Pull request merged**